### PR TITLE
Refactor ControlNetMaisi

### LINKED ISSUE #7988: Refactor MAISI networks to use monai core components
**Is your feature request related to a problem? Please describe.**

[GenerativeModels](https://github.com/Project-MONAI/GenerativeModels) is moved to monai core. We plan to refactor three MAISI networks, including AutoencoderKlMaisi, DiffusionModelUNetMaisi, and ControlNetMaisi and use monai core components.

Public
**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by guopengf**
Fixes #7988 .

### Description

Refactor ControlNetMaisi to use monai core components.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by KumoLiu** [comment]
Hi @guopengf, have you checked that the modified code can achieve the original performance? Thanks!

File: tests/test_controlnet_maisi.py [hunk 1]
```diff
@@ -103,16 +101,17 @@
 TEST_CASES_ERROR = [
     [
         {"spatial_dims": 2, "in_channels": 1, "with_conditioning": True, "cross_attention_dim": None},
-        "ControlNet expects dimension of the cross-attention conditioning "
-        "(cross_attention_dim) when using with_conditioning.",
+        "DiffusionModelUNet expects dimension of the cross-attention conditioning (cross_attention_dim) "
```
Comment (KumoLiu) [inline tests/test_controlnet_maisi.py:104]: Hi @guopengf, could you please help modify the error message in https://github.com/Project-MONAI/MONAI/blob/62430315cd176478d06dd197b7a6dfdd6cd90c44/monai/networks/nets/controlnet.py#L181?

Look like it's wrong and misleading.

Comment (guopengf) [reply tests/test_controlnet_maisi.py]: Yeah, change it to ControlNet

**Review by KumoLiu — APPROVED** [review]
Thanks for the quick update, LGTM.

**Pull request merged**